packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -827,141 +830,151 @@ function ensureRootIsScheduled(root: FiberRoot, currentTime: number) {
 // This is the entry point for every concurrent task, i.e. anything that
 // goes through Scheduler.
 function performConcurrentWorkOnRoot(root, didTimeout) {
-  if (enableProfilerTimer && enableProfilerNestedUpdatePhase) {
-    resetNestedUpdateFlag();
-  }
+  try {
+    if (enablePreloadAPIs) {
+      prepareToRender();
+    }
```

Comment:
the only diff here is wrap in `try` and add flag gated `prepareToRender` and `cleanupAfterRender` calls